rpms/SPECS/wazuh-agent.spec

```diff
@@ -35,6 +35,7 @@ log analysis, file integrity monitoring, intrusions detection and policy and com
 ./gen_ossec.sh conf agent centos %rhel %{_localstatedir} > etc/ossec-agent.conf
 
 %build
+%define initd_valid %( if [ -f /usr/lib/systemd/systemd-sysv-install ]; then echo "1" ; else echo "0"; fi )
```

Comment:
According to documentation:
```
%define has local scope, which means that it applies only to a specified part of a SPEC file. In addition, the body of a %define macro is expanded when used—​it is lazily evaluated
```
So this macro will only affect the `%build` section.

Also, the `%build` section is only executed during package generation and not during the installation so the value of this macro will be taken from the machine where the package was built.